sdk/storage/azure-storage-files-datalake/inc/azure/storage/datalake.hpp

```diff
@@ -7,4 +7,5 @@
 #include "azure/storage/files/datalake/datalake_file_client.hpp"
```

Comment:
So, if you expect there might be naming collisions with datalake or shares that are not related to files, within storage (now or in the future), it would be best to keep the `/files/` directory.

Similarly, is knowing this is related to files, as a domain-specific concept, import for the user?

```diff
-sdk/storage/azure-storage-files-datalake/inc/azure/storage/datalake.hpp
+sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake.hpp
```